src/appengine/libs/gcs.py

```diff
@@ -35,6 +36,26 @@
     ['url', 'bucket', 'key', 'google_access_id', 'policy', 'signature'])
 
 
+def service_account_email():
+  """Get the App Engine service account name."""
+  return utils.get_application_id() + '@appspot.gserviceaccount.com'
+
+
+def sign_data(data):
+  """Sign data with the default App Engine service account."""
+  iam = googleapiclient.discovery.build('iamcredentials', 'v1')
+  service_account = 'projects/-/serviceAccounts/' + service_account_email()
+
+  response = iam.projects().serviceAccounts().signBlob(
+      name=service_account,
+      body={
+          'delegates': [],
+          'payload': base64.b64encode(data),
+      }).execute()
+
+  return base64.b64decode(response['signedBlob'])
```

Comment:
Isn't the response have signature field - https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts/signBlob ?

Reply:
That one is the deprecated API :) (See top of the page).

https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/signBlob is the correct documentation.

Reply:
My bad, ignore this then.